python/lsst/ts/cbp/component.py

```diff
@@ -257,16 +262,15 @@ async def connect(self):
             await self.client.start_task
         except Exception:
             self.log.exception("Connection failed.")
-            await self.fault(code=2, report="Connection failed.")
+            await self.csc.fault(code=2, report="Connection failed.")
 
     async def disconnect(self):
         """Disconnect from the tcp socket.
 
         Safe to call even if already disconnected.
         """
-        if self.client is not None:
-            await self.client.close()
-            self.client = None
+        await self.client.close()
+        self.client = tcpip.Client(host="", port=None, log=self.log)
```

Comment:
I prefer the original implementation to be:

```python
        if self.client is not None:
            await self.client.close()
        self.client = None
```

Is there any reason that you would like to hold a new `tcpip.Client()` after the disconnection? Why not do so in the connection instead?